### ISSUE TITLE
Error when deleting fit

### ISSUE THREAD

**Issue by Lucifirius** [pyfa-org/Pyfa#1384]
```
OS version: Windows-10-10.0.16299-SP0
Python version: 3.6.1 (v3.6.1:69c0db5, Mar 21 2017, 17:54:52) [MSC v.1900 32 bit (Intel)]
wxPython version: 4.0.0b2 (wxWidgets 3.0.4)
SQLAlchemy version: 1.1.10
Logbook version: 1.0.0
Requests version: 2.11.1
Dateutil version: 2.6.0

####################

Traceback (most recent call last):
  File "C:\Users\holme\Documents\Sync\Git\blitzmann\Pyfa\gui\builtinViews\fittingView.py", line 299, in fitRemoved
  File "C:\Users\holme\Documents\Sync\Git\blitzmann\Pyfa\service\fit.py", line 1203, in refreshFit
  File "C:\Users\holme\Documents\Sync\Git\blitzmann\Pyfa\service\fit.py", line 1207, in recalc
AttributeError: 'NoneType' object has no attribute 'name'

```

Repeatable by deleting fits. Pyfa version latest Beta

**Comment by blitzmann** [comment]
Confirmed, this happens when deleting a fit that is open as far as I can tell. Fit deletion is a workflow that I want to refactor soon with the 2.x branch. But until then I should be able to simply put in a conditional here to check if fit exists before trying to rune stuff against it. Thanks for the report!

**Comment by blitzmann** [comment]
So, this is not as quick a fix as I would have hoped. pyfa has always had problems deleting fits that are currently open, and this is just another manifestation. Unfortunately, this is probably going to have to wait until I can really dive into the logic behind deleting fits and put to rest these problems for good.

**Comment by ErosianAideron** [comment]
'''
[2018-02-20 18:55:35.209225] DEBUG: service.fit: Fetching all fits for ship ID: 17726
[2018-02-20 18:55:37.992459] DEBUG: gui.builtinShipBrowser.fitItem: Deleting ship fit.
[2018-02-20 18:55:37.993461] DEBUG: service.fit: Fit::deleteFit - Deleting fit: Mission DPS NApoc (Apocalypse Navy Issue)
[2018-02-20 18:55:38.067659] DEBUG: service.fit:     Need to refresh 0 fits: set()
[2018-02-20 18:55:38.068656] WARNING: config: _+_+_+_+_+_ Fit Removed: <gui.builtinViews.fittingView.FittingView object at 0x0AE73440> True activeFitID: 9, eventFitID: 9
[2018-02-20 18:55:38.068656] DEBUG: gui.builtinViews.fittingView: FittingView::fitRemoved
[2018-02-20 18:55:38.069657] DEBUG: gui.builtinViews.fittingView:     Deleted fit is currently active
[2018-02-20 18:55:38.069657] WARNING: config: +++++ Destroy <gui.builtinViews.fittingView.FittingView object at 0x0AE73440>
[2018-02-20 18:55:38.069657] WARNING: config: True
[2018-02-20 18:55:38.070662] WARNING: config: True
[2018-02-20 18:55:38.070662] WARNING: config: True
[2018-02-20 18:55:38.070662] WARNING: config: True
[2018-02-20 18:55:38.070662] WARNING: config: True
[2018-02-20 18:55:38.096676] DEBUG: service.fit: Refresh fit for fit ID: 9
[2018-02-20 18:55:38.100678] CRITICAL: gui.errorDialog: 

Traceback (most recent call last):
  File "C:\Users\holme\Documents\Sync\Git\blitzmann\Pyfa\gui\builtinViews\fittingView.py", line 300, in fitRemoved
  File "C:\Users\holme\Documents\Sync\Git\blitzmann\Pyfa\service\fit.py", line 1203, in refreshFit
  File "C:\Users\holme\Documents\Sync\Git\blitzmann\Pyfa\service\fit.py", line 1207, in recalc
AttributeError: 'NoneType' object has no attribute 'name'
'''

**Comment by rangerwolfy** [comment]
OS version: Windows-10-10.0.16299-SP0
Python version: 3.6.1 (v3.6.1:69c0db5, Mar 21 2017, 17:54:52) [MSC v.1900 32 bit (Intel)]
wxPython version: 4.0.0b2 (wxWidgets 3.0.4)
SQLAlchemy version: 1.1.10
Logbook version: 1.0.0
Requests version: 2.11.1
Dateutil version: 2.6.0

####################

Traceback (most recent call last):
  File "C:\Users\holme\Documents\Sync\Git\blitzmann\Pyfa\gui\builtinViews\fittingView.py", line 300, in fitRemoved
  File "C:\Users\holme\Documents\Sync\Git\blitzmann\Pyfa\service\fit.py", line 1203, in refreshFit
  File "C:\Users\holme\Documents\Sync\Git\blitzmann\Pyfa\service\fit.py", line 1207, in recalc
AttributeError: 'NoneType' object has no attribute 'name'

**Comment by blitzmann** [comment]
I went ahead and implemented a stop-gap due to many reports revolving around this. It's not a proper fix, however a proper fix would be refactoring the entire deletion process. x_x

**Comment by fspigel** [comment]
Is posting more logs helpful or useless spam?

```OS version: Windows-10-10.0.16299-SP0
Python version: 3.6.1 (v3.6.1:69c0db5, Mar 21 2017, 17:54:52) [MSC v.1900 32 bit (Intel)]
wxPython version: 4.0.0b2 (wxWidgets 3.0.4)
SQLAlchemy version: 1.1.10
Logbook version: 1.0.0
Requests version: 2.18.4
Dateutil version: 2.6.0

####################

Traceback (most recent call last):
  File "C:\Users\holme\Documents\Sync\Git\blitzmann\Pyfa\gui\builtinViews\fittingView.py", line 301, in fitRemoved
  File "C:\Users\holme\Documents\Sync\Git\blitzmann\Pyfa\service\fit.py", line 1206, in refreshFit
  File "C:\Users\holme\Documents\Sync\Git\blitzmann\Pyfa\service\fit.py", line 1210, in recalc
AttributeError: 'NoneType' object has no attribute 'name'

####################

Traceback (most recent call last):
  File "C:\Users\holme\Documents\Sync\Git\blitzmann\Pyfa\gui\builtinViews\fittingView.py", line 301, in fitRemoved
  File "C:\Users\holme\Documents\Sync\Git\blitzmann\Pyfa\service\fit.py", line 1206, in refreshFit
  File "C:\Users\holme\Documents\Sync\Git\blitzmann\Pyfa\service\fit.py", line 1210, in recalc
AttributeError: 'NoneType' object has no attribute 'name'
```

**Comment by blitzmann** [comment]
>Is posting more logs helpful or useless spam?

It's always helpful, even if redundant :D Closing this now as it's been fixed with v2.0
